expo/scheduler/models.py

```diff
@@ -845,6 +848,37 @@ def class_contacts(self):
             )
         return info
 
+    def class_contacts2(self):
+        '''
+        To do: reconcile the two class_contacts functions
+        '''
+        allocations = self.resources_allocated.filter(
+            resource__in=Worker.objects.all())
+        info = []
+        for allocation in allocations:
+            try:
+                persona = allocation.resource.item.persona
+                info.append(
+                    (persona.contact_email,
+                     str(self),
+                     allocation.resource.worker.role,
+                     persona.name,
+                     persona.contact.display_name,
+                     persona.contact_phone)
+                )
+            except:
+                profile = allocation.resource.item.profile
```

Comment:
BLOCKER:  I just got an error on the current data set here.

I clicked the contact teachers link in the scheduler, this link:
http://localhost:8282/scheduler/contact_by_role/Teachers

And got:

```

DoesNotExist at /scheduler/contact_by_role/Teachers

WorkerItem has no profile.

Request Method: 	GET
Request URL: 	http://localhost:8282/scheduler/contact_by_role/Teachers
Django Version: 	1.6.11
Exception Type: 	DoesNotExist
Exception Value: 	

WorkerItem has no profile.

Exception Location: 	/usr/local/lib/python2.7/dist-packages/django/db/models/fields/related.py in __get__, line 206
Python Executable: 	/usr/bin/python
Python Version: 	2.7.6
Python Path: 	

['/vagrant/expo',
 '/home/vagrant/src/cmsplugin-image-gallery-master',
 '/usr/lib/python2.7',
 '/usr/lib/python2.7/plat-x86_64-linux-gnu',
 '/usr/lib/python2.7/lib-tk',
 '/usr/lib/python2.7/lib-old',
 '/usr/lib/python2.7/lib-dynload',
 '/usr/local/lib/python2.7/dist-packages',
 '/usr/lib/python2.7/dist-packages']

Server time: 	Tue, 2 Feb 2016 02:36:29 +0000
Traceback Switch to copy-and-paste view

    /usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py in get_response

                            response = middleware_method(request, callback, callback_args, callback_kwargs)

                            if response:

                                break

                    if response is None:

                        wrapped_callback = self.make_view_atomic(callback)

                        try:

                            response = wrapped_callback(request, *callback_args, **callback_kwargs)

         ...

                        except Exception as e:

                            # If the view raised an exception, run it through exception

                            # middleware, and if the exception middleware returns a

                            # response, use that. Otherwise, reraise the exception.

                            for middleware_method in self._exception_middleware:

                                response = middleware_method(request, e)

    ▶ Local vars
    /vagrant/expo/scheduler/views.py in contact_by_role

                header, contact_info = contact_teachers(conference)

         ...
    ▶ Local vars
    /vagrant/expo/scheduler/views.py in contact_teachers

                    contact_info += se.class_contacts2()

         ...
    ▶ Local vars
    /vagrant/expo/scheduler/models.py in class_contacts2

                        profile = allocation.resource.item.profile

         ...
    ▶ Local vars
    /usr/local/lib/python2.7/dist-packages/django/db/models/fields/related.py in __get__

                                                          self.related.get_accessor_name()))

         ...
    ▶ Local vars

Request information
GET

No GET data
POST

No POST data
FILES

No FILES data
COOKIES
Variable 	Value
csrftoken 	

'P1XIB803dpgztro2Xo2wyHxdY4gdZCHd'

sessionid 	

'ic6hr5fxjpdeq54yjts3c6jeslnewtyt'

django_language 	

'en-us'

META
Variable 	Value
wsgi.version 	

(1, 0)

RUN_MAIN 	

'true'

HTTP_REFERER 	

'http://localhost:8282/scheduler/list'

SERVER_PROTOCOL 	

'HTTP/1.1'

SERVER_SOFTWARE 	

'WSGIServer/0.1 Python/2.7.6'

SCRIPT_NAME 	

u''

LESSOPEN 	

'| /usr/bin/lesspipe %s'

SSH_CLIENT 	

'10.0.2.2 57008 22'

REQUEST_METHOD 	

'GET'

LOGNAME 	

'vagrant'

USER 	

'vagrant'

PATH 	

'/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games'

QUERY_STRING 	

''

HOME 	

'/home/vagrant'

wsgi.errors 	

<open file '<stderr>', mode 'w' at 0x7f78ffbac1e0>

LANG 	

'en_US.UTF-8'

TERM 	

'xterm-256color'

SHELL 	

'/bin/bash'

TZ 	

'UTC'

HTTP_COOKIE 	

'django_language=en-us; csrftoken=P1XIB803dpgztro2Xo2wyHxdY4gdZCHd; sessionid=ic6hr5fxjpdeq54yjts3c6jeslnewtyt'

SERVER_NAME 	

'blueprint'

REMOTE_ADDR 	

'10.0.2.2'

SHLVL 	

'1'

wsgi.url_scheme 	

'http'

SERVER_PORT 	

'8111'

CONTENT_LENGTH 	

''

XDG_RUNTIME_DIR 	

'/run/user/1000'

wsgi.file_wrapper 	

''

SSH_AUTH_SOCK 	

'/tmp/ssh-noqit53FFz/agent.5350'

CSRF_COOKIE 	

u'P1XIB803dpgztro2Xo2wyHxdY4gdZCHd'

wsgi.input 	

<socket._fileobject object at 0x7f78f36a98d0>

HTTP_USER_AGENT 	

'Mozilla/5.0 (Macintosh; Intel Mac OS X 10.9; rv:43.0) Gecko/20100101 Firefox/43.0'

HTTP_HOST 	

'localhost:8282'

wsgi.multithread 	

True

HTTP_CONNECTION 	

'keep-alive'

XDG_SESSION_ID 	

'4'

_ 	

'./manage.py'

HTTP_ACCEPT 	

'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8'

LS_COLORS 	

'rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arj=01;31:*.taz=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.zip=01;31:*.z=01;31:*.Z=01;31:*.dz=01;31:*.gz=01;31:*.lz=01;31:*.xz=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.jpg=01;35:*.jpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.axv=01;35:*.anx=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.axa=00;36:*.oga=00;36:*.spx=00;36:*.xspf=00;36:'

LESSCLOSE 	

'/usr/bin/lesspipe %s %s'

GATEWAY_INTERFACE 	

'CGI/1.1'

wsgi.run_once 	

False

SSH_TTY 	

'/dev/pts/1'

OLDPWD 	

'/home/vagrant'

wsgi.multiprocess 	

False

HTTP_ACCEPT_LANGUAGE 	

'en-US,en;q=0.5'

PWD 	

'/vagrant/expo'

DJANGO_SETTINGS_MODULE 	

'expo.settings'

CONTENT_TYPE 	

'text/plain'

MAIL 	

'/var/mail/vagrant'

SSH_CONNECTION 	

'10.0.2.2 57008 10.0.2.15 22'

REMOTE_HOST 	

''

HTTP_ACCEPT_ENCODING 	

'gzip, deflate'

PATH_INFO 	

u'/scheduler/contact_by_role/Teachers'

Settings
Using settings module expo.settings
Setting 	Value
YEAR_MONTH_FORMAT 	

'F Y'

USE_L10N 	

True

THUMBNAIL_PROCESSORS 	

('easy_thumbnails.processors.colorspace',
 'cmsplugin_nivoslider.thumbnail_processors.pad_image',
 'easy_thumbnails.processors.autocrop',
 'filer.thumbnail_processors.scale_and_crop_with_subject_location',
 'easy_thumbnails.processors.filters')

TIME_ZONE 	

'UTC'

CSRF_COOKIE_SECURE 	

False

LANGUAGE_CODE 	

'en-us'

ROOT_URLCONF 	

'expo.urls'

LOGIN_URL 	

'/login/'

BASE_DIR 	

'/vagrant/expo'

DEFAULT_CHARSET 	

'utf-8'

SESSION_SERIALIZER 	

'django.contrib.sessions.serializers.JSONSerializer'

STATIC_ROOT 	

'/vagrant/expo/gbe/static'

USE_THOUSAND_SEPARATOR 	

False

ALLOWED_HOSTS 	

[]

MESSAGE_STORAGE 	

'django.contrib.messages.storage.fallback.FallbackStorage'

EMAIL_SUBJECT_PREFIX 	

'[Django] '

SEND_BROKEN_LINK_EMAILS 	

False

SECRET_KEY 	

u'********************'

STATICFILES_FINDERS 	

('django.contrib.staticfiles.finders.FileSystemFinder',
 'django.contrib.staticfiles.finders.AppDirectoriesFinder')

SESSION_CACHE_ALIAS 	

'default'

SESSION_COOKIE_DOMAIN 	

None

SESSION_COOKIE_NAME 	

'sessionid'

ADMIN_FOR 	

()

TIME_INPUT_FORMATS 	

('%H:%M:%S', '%H:%M:%S.%f', '%H:%M')

DATABASES 	

{'default': {'ATOMIC_REQUESTS': False,
             'AUTOCOMMIT': True,
             'CONN_MAX_AGE': 0,
             'ENGINE': 'django.db.backends.mysql',
             'HOST': '',
             'NAME': 'gbe_dev',
             'OPTIONS': {},
             'PASSWORD': u'********************',
             'PORT': '',
             'TEST_CHARSET': None,
             'TEST_COLLATION': None,
             'TEST_MIRROR': None,
             'TEST_NAME': None,
             'TIME_ZONE': 'UTC',
             'USER': 'django_user'}}

FILE_UPLOAD_PERMISSIONS 	

None

FILE_UPLOAD_HANDLERS 	

('django.core.files.uploadhandler.MemoryFileUploadHandler',
 'django.core.files.uploadhandler.TemporaryFileUploadHandler')

TEMPLATE_CONTEXT_PROCESSORS 	

('django.contrib.auth.context_processors.auth',
 'django.contrib.messages.context_processors.messages',
 'django.core.context_processors.i18n',
 'django.core.context_processors.debug',
 'django.core.context_processors.request',
 'django.core.context_processors.media',
 'django.core.context_processors.csrf',
 'django.core.context_processors.tz',
 'sekizai.context_processors.sekizai',
 'django.core.context_processors.static',
 'cms.context_processors.cms_settings')

DEFAULT_CONTENT_TYPE 	

'text/html'

APPEND_SLASH 	

True

FIRST_DAY_OF_WEEK 	

0

DATABASE_ROUTERS 	

[]

CMS_STYLE_NAMES 	

(('info', 'info'),
 ('new', 'new'),
 ('hint', 'hint'),
 ('footer', 'footer'),
 ('subtitle', '20th Century Poster'),
 ('font_large', 'Large, plain font'),
 ('font_regular', 'Regular Text'))

STATICFILES_STORAGE 	

'django.contrib.staticfiles.storage.StaticFilesStorage'

CACHES 	

{'default': {'BACKEND': 'django.core.cache.backends.locmem.LocMemCache'}}

MC_API_USER 	

u'********************'

SERVER_EMAIL 	

'root@localhost'

SESSION_COOKIE_PATH 	

'/'

CMS_TEMPLATES 	

(('big_block.tmpl', 'Big Block of Content'),
 ('three_column.tmpl', 'Three Column'))

MIDDLEWARE_CLASSES 	

('django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'hijack.middleware.HijackRemoteUserMiddleware',
 'django.contrib.auth.middleware.RemoteUserMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.locale.LocaleMiddleware',
 'django.middleware.doc.XViewMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 'cms.middleware.user.CurrentUserMiddleware',
 'cms.middleware.page.CurrentPageMiddleware',
 'cms.middleware.toolbar.ToolbarMiddleware',
 'cms.middleware.language.LanguageCookieMiddleware',
 'django.contrib.flatpages.middleware.FlatpageFallbackMiddleware')

USE_I18N 	

True

THOUSAND_SEPARATOR 	

','

CMS_LANGUAGES 	

{1: [{'code': 'en-us',
      'fallbacks': [],
      'hide_untranslated': False,
      'name': 'en-us',
      'public': True,
      'redirect_on_fallback': True}],
 <class cms.utils.conf.VERIFIED at 0x7f78fb61db48>: True,
 'default': {'hide_untranslated': False,
             'public': True,
             'redirect_on_fallback': True}}

LANGUAGE_COOKIE_NAME 	

'django_language'

USER_CONTACT_RECIPIENT_ADDRESSES 	

'betty@burlesque-expo.com'

DEFAULT_INDEX_TABLESPACE 	

''

TRANSACTIONS_MANAGED 	

False

LOGGING_CONFIG 	

'django.utils.log.dictConfig'

CMSPLUGIN_FILER_IMAGE_STYLE_CHOICES 	

()

TEMPLATE_LOADERS 	

('django.template.loaders.filesystem.Loader',
 'django.template.loaders.app_directories.Loader',
 'django.template.loaders.eggs.Loader')

WSGI_APPLICATION 	

'expo.wsgi.application'

SEND_NO_MAIL 	

True

TEMPLATE_DEBUG 	

False

X_FRAME_OPTIONS 	

'SAMEORIGIN'

CSRF_COOKIE_NAME 	

'csrftoken'

CMSPLUGIN_FILER_FOLDER_DEFAULT_STYLE 	

'list'

FORCE_SCRIPT_NAME 	

None

USE_X_FORWARDED_HOST 	

False

SIGNING_BACKEND 	

'django.core.signing.TimestampSigner'

SESSION_COOKIE_SECURE 	

False

CMSPLUGIN_FILER_TEASER_STYLE_CHOICES 	

()

CSRF_COOKIE_DOMAIN 	

None

FILE_CHARSET 	

'utf-8'

DEBUG 	

True

SESSION_FILE_PATH 	

None

DEFAULT_FILE_STORAGE 	

'django.core.files.storage.FileSystemStorage'

INSTALLED_APPS 	

('cms',
 'mptt',
 'menus',
 'south',
 'sekizai',
 'djangocms_admin_style',
 'django.contrib.messages',
 'treebeard',
 'djangocms_text_ckeditor',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.admin',
 'django.contrib.sites',
 'django.contrib.sitemaps',
 'django.contrib.staticfiles',
 'django.contrib.flatpages',
 'tinymce',
 'filer',
 'easy_thumbnails',
 'image_gallery',
 'cmsplugin_nivoslider',
 'djangocms-placeholder-attr',
 'djangocms_style',
 'djangocms_column',
 'djangocms_snippet',
 'djangocms_flash',
 'djangocms_googlemap',
 'djangocms_inherit',
 'cmsplugin_filer_file',
 'cmsplugin_filer_folder',
 'cmsplugin_filer_link',
 'cmsplugin_filer_image',
 'cmsplugin_filer_teaser',
 'cmsplugin_filer_video',
 'reversion',
 'gbe',
 'ticketing',
 'scheduler',
 'django_nose',
 'hijack',
 'compat')

LANGUAGES_BIDI 	

('he', 'ar', 'fa', 'ur')

COMMENTS_ALLOW_PROFANITIES 	

False

STATICFILES_DIRS 	

('/vagrant/expo/expo/static',)

PREPEND_WWW 	

False

SECURE_PROXY_SSL_HEADER 	

None

SESSION_COOKIE_HTTPONLY 	

True

DEBUG_PROPAGATE_EXCEPTIONS 	

False

MONTH_DAY_FORMAT 	

'F j'

MANAGERS 	

()

SESSION_EXPIRE_AT_BROWSER_CLOSE 	

False

TIME_FORMAT 	

'P'

CMSPLUGIN_FILER_FOLDER_STYLE_CHOICES 	

(('list', <django.utils.functional.__proxy__ object at 0x7f78f95ef650>),
 ('slideshow', <django.utils.functional.__proxy__ object at 0x7f78f95ef6d0>))

AUTH_USER_MODEL 	

'auth.User'

DATE_INPUT_FORMATS 	

('%Y-%m-%d',
 '%m/%d/%Y',
 '%m/%d/%y',
 '%b %d %Y',
 '%b %d, %Y',
 '%d %b %Y',
 '%d %b, %Y',
 '%B %d %Y',
 '%B %d, %Y',
 '%d %B %Y',
 '%d %B, %Y')

MC_API_KEY 	

u'********************'

AUTHENTICATION_BACKENDS 	

('django.contrib.auth.backends.ModelBackend',)

EMAIL_HOST_PASSWORD 	

u'********************'

PASSWORD_RESET_TIMEOUT_DAYS 	

u'********************'

CACHE_MIDDLEWARE_ALIAS 	

'default'

SESSION_SAVE_EVERY_REQUEST 	

False

NUMBER_GROUPING 	

0

SESSION_ENGINE 	

'django.contrib.sessions.backends.db'

CSRF_FAILURE_VIEW 	

'django.views.csrf.csrf_failure'

CSRF_COOKIE_PATH 	

'/'

CMSPLUGIN_FILER_IMAGE_DEFAULT_STYLE 	

''

DECIMAL_SEPARATOR 	

'.'

IGNORABLE_404_URLS 	

()

LOCALE_PATHS 	

()

PASSWORD_HASHERS 	

u'********************'

TEMPLATE_STRING_IF_INVALID 	

''

DISALLOWED_USER_AGENTS 	

()

THUMBNAIL_HIGH_RESOLUTION 	

True

LOGOUT_URL 	

'/accounts/logout/'

EMAIL_USE_TLS 	

False

LOGIN_REDIRECT_URL 	

'/'

FIXTURE_DIRS 	

('expo/tests/fixtures',)

EMAIL_HOST 	

'localhost'

DATE_FORMAT 	

'N j, Y'

MEDIA_ROOT 	

'/vagrant/expo/media'

DEFAULT_EXCEPTION_REPORTER_FILTER 	

'django.views.debug.SafeExceptionReporterFilter'

CMSPLUGIN_FILER_FILE_DEFAULT_STYLE 	

''

FORMAT_MODULE_PATH 	

None

DEFAULT_FROM_EMAIL 	

'gbetest@burlesque-expo.com'

ADMINS 	

()

MEDIA_URL 	

'/media/'

DATETIME_FORMAT 	

'%I:%M %p'

TEMPLATE_DIRS 	

['/vagrant/expo/templates']

DATA_DIR 	

'/vagrant/expo'

SITE_ID 	

1

CMS_PERMISSION 	

True

ALLOWED_INCLUDE_ROOTS 	

()

LOGGING 	

{}

LOG_FORMAT 	

'%(asctime)s::%(levelname)s::%(funcName)s - %(message)s'

SHORT_DATE_FORMAT 	

'm/d/Y'

MC_API_ID 	

u'********************'

LOG_LEVEL 	

'INFO'

TEST_RUNNER 	

'django_nose.NoseTestSuiteRunner'

CACHE_MIDDLEWARE_KEY_PREFIX 	

u'********************'

CMSPLUGIN_FILER_FILE_STYLE_CHOICES 	

()

SOUTH_MIGRATION_MODULES 	

{'image_gallery': 'image_gallery.south_migrations'}

FILE_UPLOAD_MAX_MEMORY_SIZE 	

2621440

EMAIL_BACKEND 	

'django.core.mail.backends.smtp.EmailBackend'

DEFAULT_TABLESPACE 	

''

MC_API_URL 	

u'********************'

MIGRATION_MODULES 	

{'cms': 'cms.migrations_django',
 'djangocms_file': 'djangocms_file.migrations_django',
 'djangocms_flash': 'djangocms_flash.migrations_django',
 'djangocms_googlemap': 'djangocms_googlemap.migrations_django',
 'djangocms_inherit': 'djangocms_inherit.migrations_django',
 'djangocms_link': 'djangocms_link.migrations_django',
 'djangocms_picture': 'djangocms_picture.migrations_django',
 'djangocms_snippet': 'djangocms_snippet.migrations_django',
 'djangocms_teaser': 'djangocms_teaser.migrations_django',
 'djangocms_text_ckeditor': 'djangocms_text_ckeditor.migrations_django',
 'djangocms_video': 'djangocms_video.migrations_django',
 'menus': 'menus.migrations_django'}

SESSION_COOKIE_AGE 	

1209600

SETTINGS_MODULE 	

'expo.settings'

USE_ETAGS 	

False

LANGUAGES 	

[('en-us', 'en-us')]

FILE_UPLOAD_TEMP_DIR 	

None

INTERNAL_IPS 	

()

STATIC_URL 	

'/static/'

EMAIL_PORT 	

1025

LOG_FILE 	

'logs/main.log'

USE_TZ 	

True

SHORT_DATETIME_FORMAT 	

'm/d/Y P'

CMS_PLACEHOLDER_CONF 	

{}

ABSOLUTE_URL_OVERRIDES 	

{}

CMSPLUGIN_FILER_TEASER_DEFAULT_STYLE 	

''

TEXT_SAVE_IMAGE_FUNCTION 	

'cmsplugin_filer_image.integrations.ckeditor.create_image_plugin'

CACHE_MIDDLEWARE_SECONDS 	

600

CSRF_COOKIE_HTTPONLY 	

False

DATETIME_INPUT_FORMATS 	

('%Y-%m-%d %H:%M:%S',
 '%Y-%m-%d %H:%M:%S.%f',
 '%Y-%m-%d %H:%M',
 '%Y-%m-%d',
 '%m/%d/%Y %H:%M:%S',
 '%m/%d/%Y %H:%M:%S.%f',
 '%m/%d/%Y %H:%M',
 '%m/%d/%Y',
 '%m/%d/%y %H:%M:%S',
 '%m/%d/%y %H:%M:%S.%f',
 '%m/%d/%y %H:%M',
 '%m/%d/%y')

EMAIL_HOST_USER 	

''

PROFANITIES_LIST 	

u'********************'

You're seeing this error because you have DEBUG = True in your Django settings file. Change that to False, and Django will display a standard 500 page.
```